scripts/HPC_chatbot_preprocessor/README.md

```diff
@@ -0,0 +1,29 @@
+# Chatbot parser
+
+`chatbot_parser.py` is a script that transforms the markdown sourcefiles into a structured directory for a chatbot to be trained on. 
+
+## Generated file structure
+
+The generated directory structure is written as a subdirectory of `parsed_mds`. In `parsed_mds`, two subdirectories can be found: 
+
+- `generic` contains the parts of the markdown sources that were non-OS-specific
+- `os_specific` contains the parts of the markdown sources that were OS-specific
+
+Within `os_specific` a further distinction is made for each of the three possible operating systems included in the documentation.
+
+These subdirectories then contain a subdirectory for each individual markdown sourcefile. In the file specific subdirectories, further divisions are made according to the titles and subtitles found in that markdown sourcefile. 
+
+Finally, each of these subtitle-specific subdirectories contains a `.txt` file with the (processed) plaintext of that section and at the end a reference link to the corresponding part of the documentation website on <docs.hpc.ugent.be>.
+
```

Comment:
any chance you can also make a script (or an option) to dump the the whole structure in a json list of dicts with eg following metadata:
- text: actual text (ideally plain text, not markdown)
- url ref: url to find it
- OS: linux/windows/mac/None
- parent relation: is part of larger document. can probably be derived from url as well, but not for faq entries

Reply:
perhaps also the title as separate metadata instead of part of text